src/follower.ts

```diff
@@ -0,0 +1,100 @@
+import Resource from './resource';
+import { LinkVariables } from './types';
+
+/**
+ * The Follower class is what's being returned from follow() functions.
+ *
+ * It's 'PromiseLike', which means you can treat it like a Promise, and it
+ * can be awaited. When used as a Promise, it resolves to the Resource object
+ * that was followed.
+ *
+ * In addition to being a Promise<Resource> stand-in, it also exposes other
+ * functions, namely:
+ *
+ * * `follow()` to allow a user to chain several follow() functions to do
+ *   several 'hops' all at once.
+ * * `followAll()`, allowing a user to call `followAll()` at the end of a
+ *   chain.
+ */
+export default class Follower implements PromiseLike<Resource> {
```

Comment:
Is this going to be a real Promise?

Reply:
Using PromiseLike is the preferred approach for extending promises. It still behaves (mostly) the same and works in conjunction with other promises and async/await.

Extending the Promise object doesn't work in every browser engine. Using PromiseLike is also how for example the Bluebird library types their promises.